### PR TITLE
fix: address PR #13 Copilot review feedback

### PR DESCRIPTION
**Pull request by rnwolfe**
## Summary

Follow-up fixes for the Copilot review comments left on PR #13 (domain model, database schema, and Design CRUD API).

## Changes

1. **`server/internal/store/db.go`** — Replace the connection-scoped `PRAGMA foreign_keys=ON` exec with a DSN pragma (`?_pragma=foreign_keys(1)`) so that every connection the pool opens has FK enforcement enabled, not just the first one.

2. **`server/internal/api/handlers/designs.go`** — Wrap `r.Body` with `http.MaxBytesReader(w, r.Body, 1<<20)` (1 MB) before `json.NewDecoder` in the `Create` handler to guard against large-payload DoS attacks.

3. **`server/internal/service/design_service.go`** — Call `strings.TrimSpace(name)` before the empty-string check in `CreateDesign` so whitespace-only names (e.g. `"   "`) are rejected and the trimmed value is stored.

4. **`server/internal/store/design_store_test.go`** — Remove the brittle `isNotFound` / `containsError` string-matching helpers; replace all call sites with `errors.Is(err, models.ErrNotFound)`.

5. **`server/internal/store/migration_test.go`** — Fix the misleading `newMigrator` comment that said it returns "a migrator plus a cleanup func" — it actually returns `(*migrate.Migrate, *sql.DB)` with cleanup registered via `t.Cleanup`.

## Test plan

- [x] `cd server && go test ./...` — all 40 tests pass
- [x] `cd server && go vet ./...` — no issues

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/internal/service/design_service.go [hunk 1]
```diff
@@ -28,6 +29,8 @@ func NewDesignService(repo DesignRepository) *DesignService {
 
 // CreateDesign validates and creates a new Design.
 func (s *DesignService) CreateDesign(name, description string) (*models.Design, error) {
+	// Trim whitespace so that names like "   " are rejected alongside "".
+	name = strings.TrimSpace(name)
 	if name == "" {
 		return nil, fmt.Errorf("%w: design name is required", models.ErrConstraintViolation)
```
[the remark this answers is not in the document]
Comment (rnwolfe) [reply server/internal/service/design_service.go]: Valid — will add test cases for whitespace-only rejection and trim-before-store behavior in a follow-up.
